src/common/ceph_context.cc

```diff
@@ -425,7 +426,45 @@ void CephContext::do_command(std::string command, cmdmap_t& cmdmap,
         f->dump_string("option", *p);
       }
       f->close_section(); // unknown
-    } else if (command == "log flush") {
+    }
+    else if (command == "config diff get") {
+      std::string ceph_setting;
+      if (!cmd_getval(this, cmdmap, "var", ceph_setting)) {
+        f->dump_string("error", "syntax error: 'config diff get <var>'");
+      } //-- if (!cmd_getval(this, cmdmap, "var", ceph_setting))
+      else {
+        md_config_t def_conf;
+        def_conf.set_val("cluster", _conf->cluster);
+        def_conf.name = _conf->name;
+        def_conf.set_val("host", _conf->host);
+        def_conf.apply_changes(NULL);
+
+        map<string, pair<string, string> > diff;
+        set<string> unknown;
+        def_conf.diff_setting(_conf, &diff, &unknown, ceph_setting, true);
+        f->open_object_section("diff");
+        f->open_object_section("current");
+
+        for (map<string, pair<string, string> >::iterator p = diff.begin(); p != diff.end(); ++p) {
```

Comment:
Could we use C++11 style auto and range-based for loops instead of the older style?